### ISSUE TITLE
investigate how to support templates

### ISSUE THREAD

**Issue by sebgoa** [triggermesh/tm#125]
Say for our knative runtimes we could have a set of templates that generate a skeleton:

```
tm generate --python
```

something like https://workers.cloudflare.com/docs/quickstart/generating-a-project/

would setup a `serverless.yml` and a simple function in that language

**Comment by tzununbekov** [comment]
dev branch have `tm generate go` and `tm generate python` (there is an issue with `--` in args) commands now. Ruby and nodejs are coming.

**Comment by sebgoa** [comment]
and probably we want to give a name to it, so:

```
tm generate foo --python
```

creates a dir `foo` and puts the file in it.

**Comment by tzununbekov** [comment]
`tm generate` command with `--python`, `--go`, `--ruby` and `--node` samples is in dev. New projects being stored in corresponding dirs. I didn't add "name" parameter because it works pretty nice without it right now, but if test and see that we need it add, I do that

![image](https://user-images.githubusercontent.com/13515865/59554206-3f00fc00-8fc1-11e9-8bab-fa920ab64e69.png)

**Comment by sebgoa** [comment]
Can you improve the CLI and get some help in there, I ran into this:

```
$ ./tm generate
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/triggermesh/tm/cmd.newGenerateCmd.func1(0xc0002e3b80, 0x32437f8, 0x0, 0x0)
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/cmd/generate.go:32 +0x140
github.com/triggermesh/tm/vendor/github.com/spf13/cobra.(*Command).execute(0xc0002e3b80, 0x32437f8, 0x0, 0x0, 0xc0002e3b80, 0x32437f8)
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/triggermesh/tm/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x3219a40, 0x100552f, 0xc000042118, 0x0)
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/vendor/github.com/spf13/cobra/command.go:852 +0x2c0
github.com/triggermesh/tm/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/vendor/github.com/spf13/cobra/command.go:800
github.com/triggermesh/tm/cmd.Execute()
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/cmd/cmd.go:73 +0x2e
main.main()
	/Users/sebastiengoasguen/Documents/go/src/github.com/triggermesh/tm/main.go:22 +0x20
```

and

```
sebair-3: tm (dev)$ ./tm generate --help
2019/06/20 07:51:57 runtime "help" does not exist
```

**Comment by sebgoa** [comment]
the python template has syntax errors, mostly because it uses tabs instead of spaces...

**Comment by tzununbekov** [comment]
@sebgoa thanks for report, both issues should be fixed now

**Comment by sebgoa** [comment]
@tzununbekov we do need the ability to specify a directory, otherwise the name is hardcoded.

```
tm generate python foo
```

creates a `foo` dir with python template. Maybe one of the arg is an option depending on how cobra works

**Comment by sebgoa** [comment]
Comparing with the serverless framewor

```
sebair: Desktop $ sls create --help
Plugin: Create
create ........................ Create new Serverless service
    --template / -t .................... Template for the service. Available templates: "aws-clojure-gradle", "aws-clojurescript-gradle", "aws-nodejs", "aws-nodejs-typescript", "aws-alexa-typescript", "aws-nodejs-ecma-script", "aws-python", "aws-python3", "aws-groovy-gradle", "aws-java-maven", "aws-java-gradle", "aws-kotlin-jvm-maven", "aws-kotlin-jvm-gradle", "aws-kotlin-nodejs-gradle", "aws-scala-sbt", "aws-csharp", "aws-fsharp", "aws-go", "aws-go-dep", "aws-go-mod", "aws-ruby", "azure-nodejs", "cloudflare-workers", "cloudflare-workers-enterprise", "fn-nodejs", "fn-go", "google-nodejs", "kubeless-python", "kubeless-nodejs", "openwhisk-java-maven", "openwhisk-nodejs", "openwhisk-php", "openwhisk-python", "openwhisk-ruby", "openwhisk-swift", "spotinst-nodejs", "spotinst-python", "spotinst-ruby", "spotinst-java8", "plugin" and "hello-world"
    --template-url / -u ................ Template URL for the service. Supports: GitHub, BitBucket
    --template-path .................... Template local path for the service.
    --path / -p ........................ The path where the service should be created (e.g. --path my-service)
    --name / -n ........................ Name for the service. Overwrites the default name of the created service.



   ╭───────────────────────────────────────╮
   │                                       │
   │   Update available 1.35.0 → 1.47.0    │
   │   Run npm i -g serverless to update   │
   │                                       │
   ╰───────────────────────────────────────╯

sebair: Desktop $ sls create -t aws-python -p foo
Serverless: Generating boilerplate...
Serverless: Generating boilerplate in "/Users/sebastiengoasguen/Desktop/foo"
 _______                             __
|   _   .-----.----.--.--.-----.----|  .-----.-----.-----.
|   |___|  -__|   _|  |  |  -__|   _|  |  -__|__ --|__ --|
|____   |_____|__|  \___/|_____|__| |__|_____|_____|_____|
|   |   |             The Serverless Application Framework
|       |                           serverless.com, v1.35.0
 -------'

Serverless: Successfully generated boilerplate for template: "aws-python"
sebair: Desktop $ tree foo
foo
├── handler.py
└── serverless.yml

0 directories, 2 files
```

**Comment by tzununbekov** [comment]
@sebgoa updated cli HEAD, now it supports optional second argument:
```
> tm generate go foo

Sample go project has been created
foo/main.go		- function code
foo/serverless.yaml		- service manifest
You can deploy this project using "tm deploy -f foo --wait" command
```
